qiskit/quantum_info/operators/symplectic/stabilizer_circuit.py

```diff
@@ -0,0 +1,189 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017--2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Stabilizer to circuit function
+"""
+from __future__ import annotations
+
+from qiskit.exceptions import QiskitError
+
+from qiskit.circuit import QuantumCircuit
+
+from .clifford import Clifford
+from .pauli import Pauli
+
+
+def add_sign(stabilizer: str) -> str:
+    """
+    Add a sign to stabilizer if it is missing.
+    :param stabilizer: stabilizer string
+    :return: stabilizer string with sign
+    """
+    if stabilizer[0] not in ["+", "-"]:
+        return "+" + stabilizer
+    return stabilizer
+
+
+def drop_sign(stabilizer: str) -> str:
+    """
+    Drop sign from stabilizer if it is present.
+    :param stabilizer: stabilizer string
+    :return: stabilizer string without sign
+    """
+    if stabilizer[0] not in ["+", "-"]:
+        return stabilizer
+    if stabilizer[1] == "i":
+        return stabilizer[2:]
+    return stabilizer[1:]
+
+
+def check_stabilizers_commutator(s_1: str, s_2: str) -> bool:
+    """
+    Check if two stabilizers commute.
+    :param s_1: stabilizer string
+    :param s_2: stabilizer string
+    :return: True if stabilizers commute, False otherwise
+    """
+    prod = 1
+    for o1, o2 in zip(drop_sign(s_1), drop_sign(s_2)):
+        if o1 == "I" or o2 == "I":
+            continue
+        if o1 != o2:
+            prod *= -1
+    return prod == 1
+
+
+def apply_circuit_on_stabilizer(stabilizer: str, circuit: QuantumCircuit) -> str:
+    """
+    Given a stabilizer string and a circuit, conjugate the circuit on the stabilizer.
+    :param stabilizer: stabilizer string
+    :param circuit: Clifford circuit to apply
+    :return: stabilizer string after conjugation
+    """
+    cliff = Clifford(circuit)
+    stab_operator = Pauli(stabilizer)
+    pauli_conjugated = stab_operator.evolve(cliff, frame="s")
+    return pauli_conjugated.to_label()
+
+
+def stabilizer_to_circuit(
+    stabilizer_list: list[str],
+    allow_redundant: bool = False,
+    allow_underconstrained: bool = False,
+    invert: bool = False,
+) -> QuantumCircuit:
+    """
+    Convert a list of stabilizers to a circuit that generates state stabilized by that list using
+    Gaussian elimination with Clifford gates.
```

Comment:
isn't it simpler to just compute the destabilizers here and build a Clifford? then call Qiskit's built-in Clifford synthesis code (i.e. let qiskit synthesize it lazily in the spirit of the transpiler's high-level synthesis workflow)

Reply:
See Clifford synthesis code here: https://github.com/Qiskit/qiskit/tree/main/qiskit/synthesis/clifford

Reply:
How exactly do you propose to compute the destabilizers? The Gaussian elimination (i.e., finding the circuit generating the state) is the way I'm familiar with (see, e.g., [link from the docstring](https://quantumcomputing.stackexchange.com/questions/12721/how-to-calculate-destabilizer-group-of-toric-and-other-codes)).

Reply:
@ShellyGarion 
[`synth_clifford_greedy`](https://github.com/Qiskit/qiskit/blob/4b3316df3b8330af890ea7060f288aa7d87bb3fe/qiskit/synthesis/clifford/clifford_decompose_greedy.py#L34) appears to be similar to this code is doing, but it accepts `Clifford` rather than stabilizers, i.e., assumes that we already have the destabilizers. I do not immediately see if it can be used in this case.